pom.xml

```diff
@@ -120,6 +120,11 @@
       <scope>test</scope>
     </dependency>
     <dependency>
+      <groupId>com.damnhandy</groupId>
+      <artifactId>handy-uri-templates</artifactId>
+      <version>2.1.6</version>
```

Comment:
We’ll need to either pull this into a library plugin or use maskClasses to shield ourselves from class conflicts

Reply:
Looking at the deps: https://github.com/damnhandy/Handy-URI-Templates/blob/master/pom.xml#L473-L533

Reply:
Looking at the deps: https://github.com/damnhandy/Handy-URI-Templates/blob/master/pom.xml#L473-L533
They are all optional or test scoped except for joda-time.
If we were to put into a library plugin, this is the way to go right: https://github.com/jenkinsci/apache-httpcomponents-client-4-api-plugin with some POM magic?

Reply:
@casz yes, thats the way to go about library plugin. joda-time libraries are used at other places too. It might cause conflicts.

Reply:
@stephenc if possible, please advice if I should be doing something special against joda-time or any of the optional dependencies.

Reply:
It they are optional, then they will not be transitive.

I think joda-time may need its own api plugin, in which case you then have to add a *jenkins plugin dependency* to your api plugin... fun times

Reply:
@stephenc I give it a shot for the joda-time too :)

Reply:
Joda-time should be safe as well. However I will gladly wrap it in a plugin :) 
```xml
  <dependencies>
    <dependency>
      <groupId>org.joda</groupId>
      <artifactId>joda-convert</artifactId>
      <version>1.2</version>
      <scope>compile</scope>
      <optional>true</optional><!-- mandatory in Scala -->
    </dependency>
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>3.8.2</version>
      <scope>test</scope>
    </dependency>
  </dependencies>
```

Reply:
Currently waiting for approval: https://issues.jenkins-ci.org/projects/HOSTING/issues/HOSTING-476